File: Actividad 5/Problema 1/main.py
class Factura:
    #Atributos
    def __init__(self):
        self.numeroPieza = ""
        self.descripcion = ""
        self.cantidad = 0 
        self.precioArticulo = 0.0
        
    #Metodo
    def obtenerMontoFactura(self):
        return self.cantidad * self.precioArticulo

#Objeto
ferre = Factura()

#FacturaTest
ferre.numeroPieza = input("Ingresa el numero de pieza: ")
ferre.descripcion = input("Ingrese la descripción: ")
ferre.cantidad = int(input("Ingrese la cantidad: "))
ferre.precioArticulo = float(input("Ingrese el precio del articulo: "))

ferre.cantidad = 0 if ferre.cantidad < 0 else ferre.cantidad
ferre.precioArticulo = 0.0 if ferre.precioArticulo < 0 else ferre.precioArticulo

print("Factura:", ferre.obtenerMontoFactura())

File: Actividad 5/Problema 1/main.cpp
#include <iostream>
using namespace std;

class Factura
{
    //Atributos
    public:
        string numeroPieza;
        string descripcion;
        int cantidad;
        double precioArticulo;

    //Constructor
    Factura()
    {
        numeroPieza = "";
        descripcion = "";
        cantidad = 0;
        precioArticulo = 0.0;
    }

    //Metodos
    double obtenerMontoFactura()
    {
        return cantidad * precioArticulo;
    }
};


int main()
{
    //FacturaTest

    //Crear objeto
    Factura ferre;

    cout << "Ingresa el numero de pieza: ";
    cin >> ferre.numeroPieza;

    cout << "Ingresa la descripcion: ";
    cin >> ferre.descripcion;

    cout << "Ingresa la cantidad: ";
    cin >> ferre.cantidad;

    cout << "Ingrese el precio: ";
    cin >> ferre.precioArticulo;

    //Validaciones
    if(ferre.precioArticulo < 0)
    {
        ferre.precioArticulo = 0.0;
    }

    if(ferre.cantidad < 0)
    {
        ferre.cantidad = 0;
    }

    //Mostrar monto de la factura
    cout << "Factura: $" << ferre.obtenerMontoFactura() << endl;

    

    return 0;
}
